protocol/app/upgrades/v3.0.0/upgrade.go

```diff
@@ -1,169 +0,0 @@
-				modAccName,
-			))
-		}
-
-		// Try to get the account in state.
-		acc := ak.GetAccount(ctx, addr)
-		if acc != nil {
-			// Account has been initialized.
-			macc, isModuleAccount := acc.(sdk.ModuleAccountI)
-			if isModuleAccount {
-				// Module account was correctly initialized. Skipping
-				ctx.Logger().Info(fmt.Sprintf(
-					"module account %+v was correctly initialized. No-op",
-					macc,
-				))
-				continue
-			}
-			// Module account has been initialized as a BaseAccount. Change to module account.
-			// Note: We need to get the base account to retrieve its account number, and convert it
-			// in place into a module account.
-			baseAccount, ok := acc.(*authtypes.BaseAccount)
-			if !ok {
-				panic(fmt.Sprintf(
-					"cannot cast %v into a BaseAccount, acc = %+v",
-					modAccName,
-					acc,
-				))
-			}
-			newModuleAccount := authtypes.NewModuleAccount(
-				baseAccount,
-				modAccName,
-				perms...,
-			)
-			ak.SetModuleAccount(ctx, newModuleAccount)
-			ctx.Logger().Info(fmt.Sprintf(
-				"Successfully converted %v to module account in state: %+v",
-				modAccName,
-				newModuleAccount,
-			))
-			continue
-		}
-
-		// Account has not been initialized at all. Initialize it as module.
-		// Implementation taken from
-		// https://github.com/dydxprotocol/cosmos-sdk/blob/bdf96fdd/x/auth/keeper/keeper.go#L213
-		newModuleAccount := authtypes.NewEmptyModuleAccount(modAccName, perms...)
-		maccI := (ak.NewAccount(ctx, newModuleAccount)).(sdk.ModuleAccountI) // this set the account number
-		ak.SetModuleAccount(ctx, maccI)
-		ctx.Logger().Info(fmt.Sprintf(
-			"Successfully initialized module account in state: %+v",
-			newModuleAccount,
-		))
-	}
-}
-
-func IcaHostKeeperUpgradeHandler(
-	ctx sdk.Context,
-	vm module.VersionMap,
-	mm *module.Manager,
-) {
-	icaAppModule, ok := mm.Modules[icatypes.ModuleName].(ica.AppModule)
-	if !ok {
-		panic("Modules[icatypes.ModuleName] is not of type ica.AppModule")
-	}
-	// Add Interchain Accounts host module
-	// set the ICS27 consensus version so InitGenesis is not run
-	// initialize ICS27 module
-	vm[icatypes.ModuleName] = icaAppModule.ConsensusVersion()
-
-	// controller submodule is not enabled.
-	controllerParams := icacontrollertypes.Params{}
-
-	// host submodule params
-	hostParams := icahosttypes.Params{
-		HostEnabled:   true,
-		AllowMessages: ICAHostAllowMessages,
-	}
-
-	icaAppModule.InitModule(ctx, controllerParams, hostParams)
-}
-
-func CreateUpgradeHandler(
-	mm *module.Manager,
-	configurator module.Configurator,
-	ak authkeeper.AccountKeeper,
-) upgradetypes.UpgradeHandler {
-	return func(ctx context.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
-		sdkCtx := sdk.UnwrapSDKContext(ctx)
-		sdkCtx.Logger().Info("Running %s Upgrade...", UpgradeName)
-		InitializeModuleAccs(sdkCtx, ak)
-
-		// TODO(CORE-824): Initialize ratelimit module params to desired state.
```

Comment:
Can you move this TODO to v4.0.0 handler? ratelimit module is not implemented fully yet, but it will have genesis params and those need to be initialized properly.